### PR TITLE
fix: use ctx.Codec when marshaling for generated batch json in CLI

### PR DESCRIPTION
**Pull request by clevinson**
## Description

When reviewing #513 I noticed that we weren't using the ctx.Codec's JSON marshaler for generating batch JSON.  This is a pretty minor thing, but I figured it's good practice for us to use the same serializer as we use across other parts of the SDK, especially as we may be making use of jsonpb's configuration to serialize using the original field names (independent of explicit JSON field names defined in the proto file).

I could have written this up in an issue, but figured it'd be easier to just put in a fix.  Happy to re-evaluate this decision if I'm missing some reason why we were using the standard `json` package before.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/location.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ var reLocation = regexp.MustCompile(`^([A-Z]{2})(?:-([A-Z0-9]{1,3})(?: ([a-zA-Z0
 func validateLocation(location string) error {
 	matches := reLocation.FindStringSubmatch(location)
 	if matches == nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Invalid retirement location: %s.\nLocation should have format <country-code>[-<region-code>[ <postal-code>]].\n", location)
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Invalid location: %s.\nLocation should have format <country-code>[-<region-code>[ <postal-code>]].\n", location)
```
Comment (clevinson) [inline x/ecocredit/location.go:19]: This error gets thrown sometimes for an invalid project_location, so I updated this to make it generic to project and/or retirement location.

File: x/ecocredit/client/tx.go [hunk 2]
```diff
@@ -219,43 +223,31 @@ Required Flags:
...
+                                    "tradable_amount":     "1000",
+                                    "retired_amount":      "15",
+                                    "retirement_location": "ST-UVW XY Z12",
+                                  },
+                                ],
+                                "metadata":         "AQI=",
+                                "start_date":       "1990-01-01",
+                                "end_date":         "1995-10-31",
+                                "project_location": "AB-CDE FG1 345",
+                              }
```
Comment (clevinson) [inline x/ecocredit/client/tx.go:249]: the `Long` help text argument doesn't have access to the `*cobra.Command`, so its not possible to retrieve the sdkcontext and the JSON marshaler...  So I hardcoded an example JSON help text here.

File: x/ecocredit/location.go (hunk 1 above)
Comment (technicallyty) [reply x/ecocredit/location.go:19]: looks like a few test cases need to be updated for this

**Review by ruhatch — APPROVED** [review]
Looks great, thanks @clevinson

**Review by likhita-809 — APPROVED** [review]
lgtm

**Pull request merged**